ui/src/timeMachine/components/SubmitQueryButton.tsx

```diff
@@ -74,8 +93,24 @@ class SubmitQueryButton extends PureComponent<Props> {
     return ComponentStatus.Default
   }
 
+  private abortController: AbortController
+
   private handleClick = (): void => {
-    this.props.onSubmit()
+    // Optimistic UI, set the state to loading when the event
+    // happens rather than when the event trickles down to execution
+    this.props.setQueryToLoading()
```

Comment:
doesn't the submit action do this immediately? why add the extra iteration?

Reply:
It does, but like the comment says, we are doing this as an optimistic UI optimization. The loading state is set after the query is drafted and the query starts to execute. This is just a means of setting the expected state of the button once the button is clicked so that the perception of the action is immediate.